functions/src/sell-shares.ts

```diff
@@ -46,9 +46,28 @@ export const sellshares = newEndpoint({}, async (req, auth) => {
       throw new APIError(400, 'Trading is closed.')
 
     const prevLoanAmount = sumBy(userBets, (bet) => bet.loanAmount ?? 0)
+    const betsByOutcome = groupBy(userBets, (bet) => bet.outcome)
+    const sharesByOutcome = mapValues(betsByOutcome, (bets) =>
+      sumBy(bets, (b) => b.shares)
+    )
+
+    let chosenOutcome: 'YES' | 'NO'
+    if (outcome != null) {
+      chosenOutcome = outcome
+    } else {
+      const nonzeroShares = Object.entries(sharesByOutcome).filter(
+        ([_k, v]) => v
```

Comment:
Should we test this as `floatingEquals(v, 0)`?

Otherwise we'd have to make sure there were no floating rounding errors when you sell. And then you still could have a problem if you bought NO shares on purpose to cancel your YES shares, it might not cancel exactly even if you tried.

Reply:
You're probably right about the test here.

If you are trying to exactly cancel your shares you should be calling the API with no `amount`, and then it should use whatever amount it thinks is the exact sum of all your outstanding orders... although then if it added them up in a different order the next time you called this API, perhaps it would not add up to zero again... so yeah, anyway, you're right about this check.